src/adagrams.js

```diff
@@ -1,15 +1,136 @@
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let letters = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'];
+
+  let weights = [9, 2, 2, 4, 12, 2, 3, 2, 9, 1, 1, 4, 2, 6, 8, 2, 1, 6, 4, 6, 4, 2, 2, 1, 2, 1];
+
+  let cumulativeWeight = 98;
+
+  let hand = [];
+
+  // Repeat loop until hand is 10
+  while (hand.length < 10) {
+    // Generate a random n between 0 and total amount of available tiles
+    let randomNumber = Math.floor(Math.random() * cumulativeWeight);
+    // Iterate through letters' length
+    for (let i = 0; i < letters.length; i++) {
+      // Continously subtract current weight from cumulative weight
+      randomNumber -= weights[i];
+      // When weight reaches 0, remove letter from pool
+      if (weights[i] === 0) {
+        delete letters[i];
+      }
+      // When random n goes below 0, use current i to access the letter at that index, add letter to hand
+      if (randomNumber < 0) {
+        hand.push(letters[i]);  
+        // Subtract that tile from that letter's weight and stop subtracting
+        weights[i] -= 1;
+        break;
```

Comment:
Great work getting comfortable with JavaScript's syntax. It looks like you're getting use to looping over data and adding conditional logic ✅